components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -34,10 +38,30 @@ export default function ProjectSettingsView() {
     const { setProject } = useContext(ProjectContext);
     const { project } = useCurrentProject();
     const [showRemoveModal, setShowRemoveModal] = useState(false);
+    const [projectName, setProjectName] = useState(project?.name || "");
+    let badProjectName = projectName.length > 0 ? undefined : "Project name can not be blank.";
+    if (projectName.length > 32) {
```

Comment:
Should not we have it rather on server that it is aligned among all clients? Is it enforced by DB?